src/plugins/query_enhancements/public/query_assist/utils/create_extension.tsx

```diff
@@ -25,27 +25,30 @@ const getAvailableLanguages$ = (
   http: HttpSetup,
   data: DataPublicPluginSetup
 ) =>
-  data.query.dataSetManager.getUpdates$().pipe(
-    startWith(data.query.dataSetManager.getDataSet()),
-    distinctUntilChanged(),
-    switchMap(async (dataset) => {
-      // currently query assist tool relies on opensearch API to get index
-      // mappings, external data source types (e.g. s3) are not supported
-      if (dataset?.dataSource?.type !== DEFAULT_QUERY.ENGINE_TYPE) return [];
+  data.query.queryString
+    .getDatasetManager()
+    .getUpdates$()
+    .pipe(
+      startWith(data.query.queryString.getDatasetManager().getDataset()),
+      distinctUntilChanged(),
+      switchMap(async (dataset) => {
+        // currently query assist tool relies on opensearch API to get index
+        // mappings, external data source types (e.g. s3) are not supported
+        if (dataset?.dataSource?.type !== DEFAULT_QUERY.DATASET.DATASOURCE.TYPE) return [];
```

Comment:
Not for this PR, but we need to update this to function to  simply use the query object. Query assist should also work with indices right? This change wont work for that